### PR TITLE
Add text bumping when number is replaced w/ text

### PR DESCRIPTION
**Pull request by BeksOmega**
### Description

If a text block is currently connected to a number input, and the text changes to be not-a-number, the block will get bumped out.

This works for both the normal text block and the obfuscate text block:
NewTextBlocks

### CODE REVIEW

**Comment by ewpatton** [comment]
@BeksOmega Can you reconcile the check function with the version I wrote in #2175?

File: appinventor/blocklyeditor/src/blocks/text.js [hunk 1]
```diff
@@ -32,18 +37,47 @@ Blockly.Blocks['text'] = {
...
+  var connection = this.outputConnection.targetConnection;
+  if (!connection) {
+    return;
+  }
+  if (!isNaN(parseFloat(finalValue))) {
+    // Block is a number, so no matter where it lives it is valid.
+    return;
+  }
+
+  var typeArray = connection.check_;
```
Comment (ewpatton) [inline appinventor/blocklyeditor/src/blocks/text.js:67]: At this point, you could call `Blockly.Blocks.text.connectionCheck(this.outputConnection, this.outputConnection.otherConnection)` and check the return value. That way if we change the logic there we don't have to also change it here.

Comment (BeksOmega) [reply appinventor/blocklyeditor/src/blocks/text.js:67]: That's a really good point! Will fix

Comment (BeksOmega) [reply appinventor/blocklyeditor/src/blocks/text.js:68]: I'm going to need to move some stuff around to get this to work. Currently connectionCheck uses block.getFieldValue() to check if the field contains a number. But at the time onFinishEditing_ is called the fields value hasn't been updated yet :/ so it does still contain a number.

@ewpatton Is it ok if I wait until your hotfix is in to move stuff around? I think that'll be easier than trying to pull down your branch and rebase onto it.

Comment (ewpatton) [reply appinventor/blocklyeditor/src/blocks/text.js:68]: You could merge that branch into yours and write your code assuming it gets merged.

Comment (ewpatton) [reply appinventor/blocklyeditor/src/blocks/text.js:68]: What if we add an `opt_value` parameter to the other and if it's undefined we populate it with the block's value?

Comment (BeksOmega) [reply appinventor/blocklyeditor/src/blocks/text.js:68]: sounds good to me!

**Comment by ewpatton** [comment]
Initial code review looks good. I still need to test it, but once #2175 is merged I think this can go in.

**Comment by ewpatton** [comment]
@BeksOmega Now that #2175 has been merged, do you want to reconcile this branch?

**Comment by BeksOmega** [comment]
@ewpatton Rebased and retested :D

**Pull request merged**